packages/mdc-switch/_mixins.scss

```diff
@@ -23,18 +23,100 @@
 @import "@material/theme/mixins";
 @import "@material/ripple/mixins";
 @import "@material/rtl/mixins";
+@import "@material/elevation/mixins";
+@import "@material/feature-targeting/functions";
+@import "@material/feature-targeting/mixins";
+@import "./functions";
 @import "./variables";
 
-@mixin mdc-switch-toggled-on-color($color) {
+
+//
+// Public
+//
+@mixin mdc-switch($query: mdc-feature-all()) {
+  // postcss-bem-linter: define switch
+
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-ripple-common($query);
+
+  .mdc-switch {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-switch-base_;
+    }
+
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-switch-toggled-on-track-color($mdc-switch-baseline-theme-color);
+      @include mdc-switch-toggled-on-thumb-color($mdc-switch-baseline-theme-color);
+      @include mdc-switch-toggled-off-track-color($mdc-switch-toggled-off-track-color);
+      @include mdc-switch-toggled-off-thumb-color($mdc-switch-toggled-off-thumb-color);
+      @include mdc-switch-toggled-off-ripple-color($mdc-switch-toggled-off-ripple-color, $query);
+    }
+  }
+
+  .mdc-switch__track {
+    @include mdc-switch__track_($query);
+  }
+
+  .mdc-switch__thumb-underlay {
+    @include mdc-switch__thumb-underlay_($query);
+  }
+
+  .mdc-switch__native-control {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-switch__native-control_;
+    }
+  }
+
+  .mdc-switch__thumb {
+    @include mdc-switch__thumb_($query);
+  }
+
+  .mdc-switch--checked {
+    @include mdc-feature-targets($feat-structure) {
+      .mdc-switch__track {
+        @include mdc-switch__track-checked_;
+      }
+
+      .mdc-switch__thumb-underlay {
+        @include mdc-switch__thumb-underlay-checked_;
+      }
+
+      .mdc-switch__native-control {
+        @include mdc-switch__native-control-checked_;
+      }
+    }
+  }
+
+  .mdc-switch--disabled {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-switch--disabled-base_;
+
+      .mdc-switch__thumb {
+        @include mdc-switch__thumb-disabled_;
+      }
+
+      .mdc-switch__native-control {
+        @include mdc-switch__native-control-disabled_
+      }
+    }
+  }
+
+  // postcss-bem-linter: end
+}
+
+@mixin mdc-switch-toggled-on-color($color, $query: mdc-feature-all()) {
```

Comment:
@kfranqueiro should we add feature targeting to only the main public mixin, or these ones too? (If we are adding it here too, we should add these mixins to the test that verifies nothing is emitted for empty query, I think it would fail currently)

Reply:
That's a good point, this mixin is not invoked by the base styles, so we don't necessarily need to cover it, if the goal is specifically to add feature targeting to the base emitted styles. If you're fine with that, then I think we can omit `$query` from `mdc-switch-toggled-on-color` and `mdc-switch-toggled-off-color`.

Reply:
I could also see doing it, just for the sake of consistency on public mixins. If we always do it then people don't have to think about whether this is one of the ones that supports it.

Also for things like disabling animations, you probably do want to have the ability to exclude the `animation` feature from _all_ mixins, not just the base one.

Reply:
If we're parameterizing `$query` here, then I think we need to enclose the first 2 mixin invocations within a color feature-targets block (same for `mdc-switch-toggled-off-color`), and invoke these in `test/scss/feature-targeting.scss` as Miles suggested.